### PR TITLE
the Mug Update (?)

### PR DESCRIPTION
**Pull request by DeityLink**
imageimage
image
image

the MapDiffRenders Bot is expected to fail due to all .dmm featuring the old flag mugs being modified and some not actually compiling.

:cl:
* rscadd: Added a couple more country flag-bearing mugs. Mug boxes such as those included in Cafe Equipment supply crates now include a random one each.
* rscadd: Flag Mugs finally appear filled with liquid when some has been poured in them.
* imageadd: Re-did the in-hand icons for mugs, and added some for all flag mugs. Mugs that contain a liquid that has a unique mug sprite no longer appear invisible in-hand, and instead use the default mug sprite.

### CODE REVIEW

**Comment by SootyBear42** [comment]
Can I get a Qing era china flag cup

**Comment by Apogee-dev** [comment]
i can't believe you'd make a mug for the it*lians

**Comment by SootyBear42** [comment]
> i can't believe you'd make a mug for the it*lians

Make a cup for the Kingdom of the Two Sicilies

**Comment by DeityLink** [comment]
blankmug

~~here's a blank mug, paint the flags you want on it, I even added shading so you can use Gimp and a multiply layer to only add color.~~ 
got enough for this PR.

**Comment by SootyBear42** [comment]
> blankmug
> 
> here's a blank mug, paint the flags you want on it, I even added shading so you can use Gimp and a multiply layer to only add color.

I'm retarded

**Comment by Kurfursten** [comment]
UN mug?

image

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 1]
```diff
@@ -1170,21 +1172,68 @@
...
+/obj/item/weapon/reagent_containers/food/drinks/flagmug/francecup
+	name = "\improper cup"
+	desc = "A cup with the French flag emblazoned on it."
+	icon = 'icons/obj/cafe.dmi'
+	icon_state = "francecup"
+	volume = 30
+
+/obj/item/weapon/reagent_containers/food/drinks/flagmug/italycup
+	name = "\improper cup"
+	desc = "A cup with the Italy flag emblazoned on it."
```
Comment (BomberBro) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: Italian flag

**Comment by PrimeDSS13** [comment]
remove the french mug immediately

**Comment by Zth--** [comment]
arg
uru

**Comment by DeityLink** [comment]
> 
> 
> remove the french mug immediately

Never!

**Comment by DeityLink** [comment]
> 
> 
> UN mug?
> 
> image

I also added the EU flag

**Comment by Eneocho** [comment]
fucking argentinian mug
enjoy the recession

**Comment by Eneocho** [comment]
Also, instead of adding more mugs, this should be a source of the rarer mugs, like the gromit mug.

**Comment by DeityLink** [comment]
Are there any other rare mugs beside the gromit mug?

**Comment by help-maint** [comment]
> 
> 
> arg
> uru

Why did you sprite the Argentina mug twice?

**Comment by help-maint** [comment]
Also, maybe instead of having a flagmug lootbox, you could paint a flag onto a blank mug via RSF/tilepainter?

**Comment by DeityLink** [comment]
I think it'd remove from the collectible aspect.

**Comment by help-maint** [comment]
I mean are they really collectible though

**Comment by DeityLink** [comment]
They are now.

**Comment by DeityLink** [comment]
I also improved my old mug in-hand and made some new ones for all flag mugs
image

**Comment by SonixApache** [comment]
Hey, I was going to do this two years ago but forgot, good stuff. :+1:
Since you're adding the other flags, I'd say replace the american flag in the priority rewards box with the random flag mug too.

**Comment by DeityLink** [comment]
> 
> 
> Hey, I was going to do this two years ago but forgot, good stuff. 👍
> Since you're adding the other flags, I'd say replace the american flag in the priority rewards box with the random flag mug too.

I was wondering about why it was an american mug specifically, will do.

**Comment by DeityLink** [comment]
there

**Comment by Eneocho** [comment]
> 
> 
> blankmug
> 
> here's a blank mug, paint the flags you want on it, I even added shading so you can use Gimp and a multiply layer to only add color.

chilemug2
chilemug

**Comment by Eneocho** [comment]
now we need someone to make all fucking countries

**Comment by BomberBro** [comment]
Are you a bad enough dude to use _these_ mugs?
raremugs

**Comment by DeityLink** [comment]
> 
> 
> Are you a bad enough dude to use _these_ mugs?
> raremugs

are you from any of those countries? I can't recognize the flags. The point is mostly to let players have their home country mug, not have every single country in there because otherwise this PR is never getting finished.

**Comment by Eneocho** [comment]
From left to right they look like:
* Sahrawi Arab Democratic Republic's flag.
* Transnistria's flag.
* Armenia's flag (though that white patch isn't on the real flag, did he accidentally not paint over it?).
* Taiwan's flag.

If we gonna meme flags, we also need palestine and nazi germany's flag mugs.

**Comment by DeityLink** [comment]
Yeah nah, that wasn't the point of this PR, I just thought I'd add a few ones from countries that our players come from. Let's not go down the slippery slope to nazi mugs and kekistan mugs. or god forgive, israel/palestinian mugs

**Comment by Eneocho** [comment]
> 
> 
> Yeah nah, that wasn't the point of this PR, I just thought I'd add a few ones from countries that our players come from. Let's not go down the slippery slope to nazi mugs and kekistan mugs. or god forgive, israel/palestinian mugs

Even if we were to go down the meme mug route, it'd be hard as hell to sprite it. How the fuck do you sprite a swastika in a 3x3 area. Same with kekistan.
Palestine's flag would be easy, it's literally the same as the arab democratic republic's flag but without the crescent moon.

**Comment by DeityLink** [comment]
> Even if we were to go down the meme mug route, it'd be hard as hell to sprite it. How the fuck do you sprite a swastika in a 3x3 area. Same with kekistan.

I managed with the EU flag. As long as the correct colors are used, the brain is able to extrapolate meaning from a mess of pixel.

**Pull request merged**